example/fine_advection/advection_driver.cpp

```diff
@@ -115,37 +132,31 @@ TaskCollection AdvectionDriver::MakeTaskCollection(BlockList_t &blocks, const in
                             desc_fine, face, parthenon::CellLevel::fine, mc0.get());
     }
 
-    auto set_flx = parthenon::AddFluxCorrectionTasks(start_flxcor | flx | flx_fine, tl,
-                                                     mc0, pmesh->multilevel);
-
-    auto flux_div = tl.AddTask(set_flx, Stokes<pack_desc_t>, parthenon::CellLevel::same,
-                               parthenon::TopologicalType::Cell, desc, pmesh->ndim,
-                               mc0.get(), mdudt.get());
-
-    auto flux_div_fine = tl.AddTask(
-        set_flx, Stokes<pack_desc_fine_t>, parthenon::CellLevel::fine,
-        parthenon::TopologicalType::Cell, desc_fine, pmesh->ndim, mc0.get(), mdudt.get());
+    auto vf_dep = none;
+    for (auto edge : std::vector<TE>{TE::E1, TE::E2, TE::E3}) {
+      vf_dep = tl.AddTask(vf_dep, advection_package::CalculateVectorFluxes<C, D>, edge,
```

Comment:
I dislike re-using TaskIDs like this (does this actually all work okay under the hood in the tasking infrastructure?).  I understand that this situation is a bit unique in that the for loop is called over a subset of topological elements, but is there a path here that doesn't involve re-using `vf_dep`?

Reply:
I think there is no issue under the hood with re-initializing `TaskID` objects with new task identifiers. 

Although there is probably a way to not reuse the task ID here, it seems like it might be over kilI to change things. Maybe take that with a grain of salt, since I personally like the pattern of reusing TaskID objects as much as possible since it makes it easier to add and remove tasks with sequential dependency when debugging or looping as above.